advanced_samples/CSV/dotnet/CSVtoOCS_pkce/Program.cs

```diff
@@ -0,0 +1,300 @@
+                    var stream2 = new SdsStream { Id = stream2ID, TypeId = typeToCreate.Id };
+                    Console.WriteLine("Creating Stream");
+                    stream1 = await metaService.GetOrCreateStreamAsync(stream1);
+                    stream2 = await metaService.GetOrCreateStreamAsync(stream2);
+                }
+
+                Console.WriteLine("Sending Data");
+                // Loop over each stream to send to and send the data as one call.
+                foreach (string streamId in streamsIdsToSendTo)
+                {
+                    // Get all of the data for this stream in a list
+                    var valueToSend = dataList.Where(dataEntry => dataEntry.StreamId == streamId)  //gets only appropriate data for stream
+                                              .Select(dataEntry => new TemperatureReadings(dataEntry)) // transforms it to the right data
+                                              .ToList(); // needed in IList format for insertValues
+                    await dataService.InsertValuesAsync(streamId, valueToSend);
+                }
+
+                if (test)
+                {
+                    //checks to make sure values are written
+                    await CheckValuesWrittenASync();
+                }
+            }
+            catch (Exception ex)
+            {
+                success = false;
+                Console.WriteLine(ex.Message);
+                toThrow = ex;
+            }
+            finally
+            {
+                if (test)
+                {
+                    if (!createStreams)
+                    {
+                        // if we just created the data lets just remove that
+                        // Do Delete
+                        Console.WriteLine("Deleting Data");
+                        await DeleteValuesAsync();
+                        // Do Delete check
+                        await CheckDeletesValuesAsync();
+                    }
+                    else
+                    {
+                        Console.WriteLine("Deleting Streams");
+                        // if we created the types and streams, lets remove those too
+                        await RunInTryCatch(metaService.DeleteStreamAsync, stream1ID);
+                        await RunInTryCatch(metaService.DeleteStreamAsync, stream2ID);
+                        Console.WriteLine("Deleting Types");
+                        await RunInTryCatch(metaService.DeleteTypeAsync, typeID);
+
+                        // Check deletes
+                        await RunInTryCatchExpectException(metaService.GetStreamAsync, stream1ID);
+                        await RunInTryCatchExpectException(metaService.GetStreamAsync, stream2ID);
+                        await RunInTryCatchExpectException(metaService.GetTypeAsync, typeID);
+                    }
+                }
+            }
+
+            if (toThrow != null)
+                throw toThrow;
+            return success;
```

Comment:
I've been deleting the `success` variable since `success` is always the same as `toThrow == null`.